pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/TaskGeneratorUtils.java

```diff
@@ -68,4 +73,30 @@
     }
     return runningSegments;
   }
+
+  public static Map<String, List<String>> getScheduledSegmentsMap(@Nonnull String taskType,
```

Comment:
There's no side effect. We filter out segments that are already scheduled. So, if the task is completed right after we fetch the state from the helix, the segment will be available for scheduling on the next round. 

```
Can this return segments that are no longer scheduled?
```
I'm not quite sure what will this would improve?

What we need to do is:

1. fetch all the segments from the table
2. filter out segments that are already scheduled for merge tasks